static/src/javascripts/projects/journalism/services/impl/viewport-io.js

```diff
@@ -0,0 +1,61 @@
+// @flow
```

Comment:
is this checking whether atoms are in view?

Reply:
can you explain what `impl` means here?

Reply:
Yep. `viewport` is the abstract service, there are two implementations based on browser support: one using `IntersectionObserver`, the other using a scroll listener.